pubtools/pulplib/_impl/model/repository/file.py

```diff
@@ -20,3 +29,79 @@ class FileRepository(Repository):
     )
 
     mutable_urls = attr.ib(default=attr.Factory(lambda: ["PULP_MANIFEST"]), type=list)
+
+    def upload_file(self, file_obj, relative_url=None):
+        """Upload a file to this repository.
+
+        Args:
+            file_obj (str, file object)
+                If it's a string, then it's the path of file to upload.
+                Else, it ought to be a
+                `file-like object <https://docs.python.org/3/glossary.html#term-file-object>`_.
+
+
+            relative_url (str)
+                Path that should be used in remote repository, can either
+                be a path to a directory or a path to a file, e.g:
+                - if relative_url is 'foo/bar/' and file_obj has name 'f.txt',
+                  the resulting remote path wll be 'foo/bar/f.txt'.
+                - if relative_url is 'foo/bar/f.txt', no matter what the
+                  name of file_obj is, the remote path is 'foo/bar/f.txt'.
+
+                If omitted, the local name of the file will be used. Or,
+                if file_obj is a file object without a `name` attribute,
+                passing `relative_url` is mandatory.
+
+        Returns:
+            Future[:class:`~pubtools.pulplib.Task`]
```

Comment:
Documentation maybe doesn't match reality again - isn't it returning a future with a *list of tasks*, not a single task?

Reply:
copy-pasted and wrong again :(